File: src/main/java/cz/wa/secretagent/io/campaign/model/ExplosionModelCreator.java
package cz.wa.secretagent.io.campaign.model;

import java.util.List;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import cz.wa.secretagent.io.tiles.singleproperties.ModelProperties;
import cz.wa.secretagent.view.TileId;
import cz.wa.secretagent.worldinfo.graphics.ModelInfo;
import secretAgent.view.model.ExplosionModel;
import secretAgent.world.GLModel;

/**
 * Creates explosion model. 
 * 
 * @author Ondrej Milenovsky
 */
public class ExplosionModelCreator extends AbstractModelCreator {
    private static final long serialVersionUID = 4746311062870890398L;

    private static final Logger logger = LoggerFactory.getLogger(ExplosionModelCreator.class);

    private static final String FRAMES_PROPERTY = "frames";

    @Override
    public ExplosionModel createModel(ModelProperties modelProperties, TileId tileId, String modelName) {
        // frames
        List<TileId> tileIds = modelProperties.getProperties().get(FRAMES_PROPERTY);
        if (tileIds == null) {
            logger.warn("Explosion model '" + modelName + "' must have property: " + FRAMES_PROPERTY);
            return null;
        }
        return new ExplosionModel(tileIds);
    }

    @Override
    public ModelInfo createModelInfo(TileId tileId, GLModel model, String modelName) {
        return new ModelInfo(0, model);
    }

}

File: src/main/kotlin/secretAgent/io/campaign/model/ExplosionModelCreator.kt
package secretAgent.io.campaign.model

import cz.wa.secretagent.worldinfo.graphics.ModelInfo
import org.slf4j.LoggerFactory
import secretAgent.io.tiles.ModelProperties
import secretAgent.view.model.ExplosionModel
import secretAgent.view.renderer.TileId
import secretAgent.world.GLModel

/**
 * Creates explosion model.
 *
 * @author Ondrej Milenovsky
 */
class ExplosionModelCreator : AbstractModelCreator() {

    override fun createModel(modelProperties: ModelProperties, tileId: TileId, modelName: String): ExplosionModel? {
        // frames
        val tileIds = modelProperties.properties[FRAMES_PROPERTY] ?: run {
            logger.warn("Explosion model '$modelName' must have property: $FRAMES_PROPERTY")
            return null
        }
        return ExplosionModel(tileIds)
    }

    override fun createModelInfo(tileId: TileId, model: GLModel, modelName: String) = ModelInfo(0, model)

    companion object {
        private const val serialVersionUID = 4746311062870890398L

        private val logger = LoggerFactory.getLogger(ExplosionModelCreator::class.java)

        private const val FRAMES_PROPERTY = "frames"
    }
}